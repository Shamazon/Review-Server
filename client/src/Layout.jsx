import React from 'react';
import Reviews from './components/App.jsx';
import Description from '../../proxy/Description/src/components/App.jsx'
import Photo from '../../proxy/Photo-Gallery/client/src/components/app.jsx'
import Related from '../../proxy/Related-Items/client/src/components/App.jsx'
import style from '../dist/style/layout.css'

const Layout = (props) => {
  return (
    <div>
      <div className={style.header}>
      </div>
      <div className={style.descriptionContainer} >
        <div className={style.photo} >
          <Photo id={props.params.id} />
        </div>
        <div className={style.description}>
          <Description id={props.params.id} />
          <div className={style.boughtTogether} >
          </div>
        </div>
        <div className={style.buyPhoto}>
        </div>
      </div>
        <Related id={props.params.id} />
        <div className={style.questions} >
        </div>
        <Reviews product={props.params.id} />
        <Related id={props.params.id} />
        <div className={style.footer} >
        </div>
    </div>
  )
}

export default Layout;