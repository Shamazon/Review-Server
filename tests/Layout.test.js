import React from 'react';
import Layout from '../client/src/Layout.jsx';
import enzyme from 'enzyme';

describe('Layout', () => {
  it('should be defined', () => {
    expect(Layout).toBeDefined();
  });
  it('should render correctly', () => {
    const props = {
      params: {
        id: 0,
      },
    };
    const tree = enzyme.shallow(<Layout {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
