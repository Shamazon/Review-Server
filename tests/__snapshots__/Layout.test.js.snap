// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Layout
    params={
      Object {
        "id": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header"
        />,
        <div
          className="descriptionContainer"
        >
          <div
            className="photo"
          >
            <App
              id={0}
            />
          </div>
          <div
            className="description"
          >
            <App
              id={0}
            />
            <div
              className="boughtTogether"
            />
          </div>
          <div
            className="buyPhoto"
          />
        </div>,
        <App
          id={0}
        />,
        <div
          className="questions"
        />,
        <App
          product={0}
        />,
        <App
          id={0}
        />,
        <div
          className="footer"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "header",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="photo"
            >
              <App
                id={0}
              />
            </div>,
            <div
              className="description"
            >
              <App
                id={0}
              />
              <div
                className="boughtTogether"
              />
            </div>,
            <div
              className="buyPhoto"
            />,
          ],
          "className": "descriptionContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <App
                id={0}
              />,
              "className": "photo",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <App
                  id={0}
                />,
                <div
                  className="boughtTogether"
                />,
              ],
              "className": "description",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "id": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "boughtTogether",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "buyPhoto",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "questions",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "product": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "footer",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header"
          />,
          <div
            className="descriptionContainer"
          >
            <div
              className="photo"
            >
              <App
                id={0}
              />
            </div>
            <div
              className="description"
            >
              <App
                id={0}
              />
              <div
                className="boughtTogether"
              />
            </div>
            <div
              className="buyPhoto"
            />
          </div>,
          <App
            id={0}
          />,
          <div
            className="questions"
          />,
          <App
            product={0}
          />,
          <App
            id={0}
          />,
          <div
            className="footer"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "header",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="photo"
              >
                <App
                  id={0}
                />
              </div>,
              <div
                className="description"
              >
                <App
                  id={0}
                />
                <div
                  className="boughtTogether"
                />
              </div>,
              <div
                className="buyPhoto"
              />,
            ],
            "className": "descriptionContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <App
                  id={0}
                />,
                "className": "photo",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "id": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <App
                    id={0}
                  />,
                  <div
                    className="boughtTogether"
                  />,
                ],
                "className": "description",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "id": 0,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "boughtTogether",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "buyPhoto",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "questions",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "product": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "footer",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
